test/test-expression-side-effect.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as stackdriver from '../src/types/stackdriver';
+import * as debugapi from '../src/agent/v8/debugapi';
+import * as extend from 'extend';
+import {defaultConfig} from '../src/agent/config';
+import * as scanner from '../src/agent/io/scanner';
+import * as SourceMapper from '../src/agent/io/sourcemapper';
+import {Common, LoggerOptions} from '../src/types/common';
+import * as assert from 'assert';
+import {StatusMessage} from '../src/client/stackdriver/status-message';
+import * as utils from '../src/agent/util/utils';
+
+const code = require('./test-expression-side-effect-code.js');
+
+const common: Common = require('@google-cloud/common');
+
+const nodeAtLeast8withInspector = utils.satisfies(process.version, '>=8') && process.env.GCLOUD_USE_INSPECTOR;
+const nodeAtLeast8OrInspector = utils.satisfies(process.version, '>=8') || process.env.GCLOUD_USE_INSPECTOR;
+const nodeAtLeast9withInspector = utils.satisfies(process.version, '>=9.1.0') && process.env.GCLOUD_USE_INSPECTOR;
```

Comment:
I am not sure if all these combinations still make sense. My expectation is that `GCLOUD_USE_INSPECTOR` should be the default for Node 10+. I would prefer that `GCLOUD_USE_INSPECTOR` go away. That is not this PR though, but there is complexity in this one because of that env var.

Can we remove that env. var. in a separate PR and rebase this on top?

Reply:
Sure